var/spack/repos/builtin/packages/mvapich2-gdr/package.py

```diff
@@ -71,12 +71,33 @@ class Mvapich2Gdr(AutotoolsPackage):
         default=False
     )
 
+    variant(
+        'cuda',
+        description='Enable/Disable support for cuda',
+        default=True
+    )
```

Comment:
Does it make sense to depend on `CudaPackage` to enable cuda support? Same question for Rocm.

Reply:
I've been told that since gdr is only available through selectively available binaries, when a user does `spack buildcache list` they won't be able to see which hash is for a cuda build and which is for rocm. 
 @nithintsk has made a feature request to address this, it seems https://github.com/spack/spack/issues/17998

Reply:
Ok, thanks for the pointer to #17998 which fell definitely off my radar! What I am asking though is not to remove or change the variant, but if it makes sense to inherit it via the `CudaPackage` base class (which does also other things that may be wanted / not wanted depending on the package).

Reply:
Hmm I hadn't considered that. I am pretty much brand new to the Spack ecosystem and don't quite know the lay of the land. I'll take a look at the docs, but my naive question is that if that would be a problem, since there is also a rocm variant now? And would that cause problems, since (to my knowledge), they are mutually exclusive?

Reply:
Upon looking at the docs, I think it is a good suggestion, but it does not seem appropriate right now for a couple of reasons. The first is that for GDR I think we want building with cuda to be enabled by default (although I guess I could dig into it and find a way to override `CudaPackage`'s default behaivor). I also do not think we do anything with CUDA architectures at  this point, but perhaps @nithintsk can comment.